gammapy/analysis/core.py

```diff
@@ -274,320 +215,24 @@ def write_datasets(self, overwrite=True, write_covariance=True):
                                  filename_models,
                                  overwrite=overwrite,
                                  write_covariance=write_covariance)
-            log.info(f"Datasets stored to {filename}.")
-            log.info(f"Datasets stored to {filename_models}.")
+            self.log.info(f"Datasets stored to {filename}.")
+            self.log.info(f"Datasets stored to {filename_models}.")
         else:
             raise RuntimeError("Missing datasets_file in config.general")
 
 
-    def run_fit(self):
-        """Fitting reduced datasets to model."""
-        if not self.models:
+    def check_datasets(self):
```

Comment:
is dataset reading and/or model creation and setting a specific analysis step?

Reply:
no but it could be, for now reading of config.general.datasets_file is done through the data-selection step if the file exists, and if overwrite is False.